### ISSUE TITLE
Cuda streams in WorkGraphPolicy

### ISSUE THREAD

**Issue by berselius** [kokkos/kokkos#3230]
Requesting an addition of cuda streams to the `WorkGraphPolicy`. Would be great to give this a list of streams that could launch the tasks in e.g. a round robin fashion across the streams. 

Also this could be a good time to add hierarchical parallelism to WorkGraphPolicy :D (#3045)

**Comment by jrmadsen** [comment]
Good idea. IMO should probably support a functor for how the streams are selected, with the default being the round-robin.
